processing/src/main/java/org/apache/druid/java/util/common/guava/ParallelMergeCombiningSequence.java

```diff
@@ -1310,9 +1327,22 @@ public long getSlowestPartitionInitializedTime()
 
     private final int inputSequences;
 
+    public enum State
+    {
+      SCHEDULED,
+      STARTED,
+      PARTITON_MERGE_SCHEDULED,
+      FINAL_MERGE_SCHEDULED
+    }
```

Comment:
could you explain the states a bit better? Like, the tasks from both layers of the merge are all started at basically the same time, and keep scheduling additional actions if there is more stuff to process until their inputs are completely drained. So I don't really understand how the state transitions are supposed to work and what they are supposed to tell us. It seems like unless it gets blocked completely at SCHEDULED, chances are it will burn through the other two scheduled states like immediately (since its just adding tasks to a queue, not blocked on them actually running afaict). Im not really sure there are meaningful states between haven't started, running, and done, and done isn't very interesting. More interesting seems like the information about the number of tasks each partition has run and/or number of rows they have processed, vs amount of stuff processed by the 2nd layer "final merge" tasks (like is it being held up by a slow partition, etc), but I'm not really sure how to easily wrap that stuff up in state.